### PR TITLE
refactor perform task

### PR DESCRIPTION
**Pull request by MarinPostma**
Refactor some index resolver functions to make duties more consistent, and code easier to test

### CODE REVIEW

File: meilisearch-lib/src/index_resolver/mod.rs [hunk 1]
```diff
@@ -96,349 +102,454 @@ impl FromStr for IndexUid {
...
+                TaskContent::DocumentDeletion {
+                    deletion: DocumentDeletion::Ids(ids),
+                    index_uid,
+                } => {
+                    let ids = ids.clone();
+                    let index = self.get_index(index_uid.clone().into_inner()).await?;
+
+                    let DocumentDeletionResult {
+                        deleted_documents, ..
+                    } = spawn_blocking(move || index.delete_documents(&ids)).await??;
```
Comment (irevoire) [inline meilisearch-lib/src/index_resolver/mod.rs:255]:
```suggestion
                TaskContent::DocumentDeletion {
                    deletion: DocumentDeletion::Ids(ref ids),
                    index_uid,
                } => {
                    let index = self.get_index(index_uid.clone().into_inner()).await?;

                    let DocumentDeletionResult {
                        deleted_documents, ..
                    } = spawn_blocking(move || index.delete_documents(&ids)).await??;
```

Wouldn’t that work? (I removed the ids.clone)

**Review by irevoire — APPROVED** [review]
lgtm good job 👍

File: meilisearch-lib/src/index_resolver/mod.rs (hunk 1 above)
Comment (MarinPostma) [reply meilisearch-lib/src/index_resolver/mod.rs:255]: no, that woudn't work since I only have a ref here, it would need to be static

**Comment by MarinPostma** [comment]
waiting for #2466

**Comment by MarinPostma** [comment]
bors merge

**Comment by curquiza** [comment]
bors merge

**Pull request merged**